### DISCUSSION TITLE
Discuss the influence of 'Scheduler' and 'Extract EMA Weights' when creating a model

### DISCUSSION THREAD

**Discussion by lyuro** [d8ahazard/sd_dreambooth_extension#860] [General]
***First of all, this article uses Google translation, I hope it is readable***

When I create a Dreambooth model, I always have doubts, that is, what parameters should be selected to create the model. After all, it is the first step
Because I want to do a quick comparison, I only use 1 image, and use the training configurator **(object/style)** which is 100 steps for testing.
Use commonly used Any3 7g, NAI7G, ACertainThing as source models,
Use DDIM, Euler-ancestral, whether to check Extract EMA Weights, a total of 12 samples
[filewords] used during training.

The following are some generated results, hoping to serve as some reference

Prompt:
 Houkai3rd, 1girl,raiden mei, 
Seed: 
 420420 , 20230126
Sampler: 
Euler a
Sampling steps:
30
CFG Scale:
7.5

Any3 7G
![xyz_grid-0015-420420-Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214681626-2b3890bb-b9ce-404c-943c-afa36159362b.png)
![xyz_grid-0016-20230126-Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214681631-6f18a089-b3ab-4d36-b0b1-1b95cf2c0056.png)

ACertainThing 
![xyz_grid-0013-420420-Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214681914-93f660f8-6628-4576-a65e-ec56064ed44c.png)
![xyz_grid-0017-20230126-Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214681932-5250cd7a-31c8-4f43-bb3d-39a6e7fbbaa0.png)

NAI 7G
![xyz_grid-0014-420420-Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214682363-93b5d683-9c77-489a-99e8-c01c3725b8c9.png)
![xyz_grid-0018-20230126-Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214682372-50cd924f-18c2-4a46-a22a-574a9d860b93.png)

---------Oh I actually forgot the output of the source model for comparison----------

![xyz_grid-0023-168114106-masterpiece, best quality, delicate face, highres, extremely detailed 8k wallpaper,_Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214686333-c16119ce-7042-44cf-9e8e-b56bc737c52d.png)
![xyz_grid-0024-168114106-masterpiece, best quality, delicate face, highres, extremely detailed 8k wallpaper,_Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214686325-10dbd2b6-72e4-45d6-ac64-1aeaf75edcfe.png)
![xyz_grid-0022-168114106-masterpiece, best quality, delicate face, highres, extremely detailed 8k wallpaper,_Houkai3rd, 1girl,raiden mei,](https://user-images.githubusercontent.com/38397800/214686338-8df9ac18-3383-41cb-9004-8e5884690faf.png)

Would be grateful if someone summarizes the characteristics of each option and leaves a message

**Comment by Zuxier** [comment]
In theory, DDIM without EMA extraction should be the way to go for resuming training.
That said, not much test have been done on those option, so if you wish explore them to seek better training feel free.
The options are there because they are exposed.
DDIM is native to the models, EMA is there to assist images generation.

**Reply by giteeeeee** [reply]
Then if I would extract the EMA out during training, how am I going to bake the EMA back in when the training is done since it helps with image generation?

**Reply by Zuxier** [reply]
> Then if I would extract the EMA out during training, how am I going to bake the EMA back in when the training is done since it helps with image generation?

i usually don't extract ema (that from what i remember reading is bad for training)
but i usually select the use ema option (it uses more vram, you can only use it for 16+ GB cards) and i feel it improves the final model.
I m happy of my training results so i never looked into it too much.

**Reply by lyuro** [reply]
I tried the use ema option, but I found it generally not as good as not using it.
(style and character)
I don't know if it's a matter of style

**Comment by giteeeeee** [comment]
Thanks for the comparison. Though I often find it almost impossible to test models with small-moderate difference. Simply because even if you train the same model with the same parameters, they will generate mildly different images. Really hard to control for parameters of small impact.
That said, I would like to have the EMA option discussed as I have no clue what it does.

**Comment by cerega66** [comment]
There is very little information about the EMA, especially good information, so I collected it in parts. EMA is something like a dataset of previous trainings with their results. Used to continue trainings so you don't have to start trainings from the beginning. And I don’t think that this is of real benefit in our case, since the principle of DB trainings can hardly be called a continuation. I also think it makes sense to most that it makes no sense to use this option on non-EMA models.

You can read about the scheduler here https://arxiv.org/pdf/2206.00364.pdf

But in short, yes, it affects training, but it’s hard to say by how much and how. And about which one is better, which one is worse is even more difficult. They just implement different algorithms that lead to the same results. Therefore, I use the default scheduler offered in diffusers: PNDM.

**Reply by cerega66** [reply]
From one help: "EMA weights usually yield higher quality images for inference. Non-EMA weights are usually better to continue fine-tuning."
